core/plugin_api.rs

```diff
@@ -11,13 +11,31 @@
 pub use crate::Buf;
 pub use crate::Op;
 pub use crate::OpId;
+pub use crate::Resource;
+pub use crate::ResourceId;
+pub use crate::ResourceTable;
 pub use crate::ZeroCopyBuf;
 
 pub type InitFn = fn(&mut dyn Interface);
 
 pub type DispatchOpFn =
   fn(&mut dyn Interface, &[u8], Option<ZeroCopyBuf>) -> Op;
 
+/// Equivilent to ResourceTable for normal ops, but only includes
+/// non generic("boxed") versions of get, get_mut, and remove.
+/// We can't have generic functions on a trait and use it as a object
+/// this is a limtation of rust.
+pub trait WrappedResourceTable {
```

Comment:
Why do all these methods have `_boxed` in their name? 
There's only a few that actually do...

Reply:
BTW I think the function signatures are good, the names are just odd...

Reply:
I had originally suffixed them with `boxed`, because they returned a `&Box<dyn Resource>` value. Turns out the `Box` usage is unneeded. Not sure how to differentiate these from their counterparts. Maybe use `dyn` or `raw` suffix?